### PR TITLE
[FEM-DEM] Minor improvements in the FEMDEM elements

### PR DESCRIPTION
**Pull request by AlejandroCornejo**
In this Pr I'm correcting the CalculateLHS method by computing the previous step Secant Matrix instead of the updated tangent one.

### CODE REVIEW

File: applications/FemToDemApplication/custom_elements/femdem2d_element.cpp [hunk 1]
```diff
@@ -386,7 +344,7 @@ void FemDem2DElement::CalculateRightHandSide(VectorType& rRightHandSideVector, P
 	Flags& ConstitutiveLawOptions = values.GetOptions();
 
 	ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_STRESS);
-	ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR);
+	//ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR);
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_elements/femdem2d_element.cpp]: Remove it if unused
```suggestion
```

File: applications/FemToDemApplication/custom_elements/femdem2d_element.cpp [hunk 2]
```diff
@@ -1469,4 +1429,162 @@ void FemDem2DElement::AssignComponentsToSecondOrderCentralDifferencesTangentTens
...
+	Vector current_displacements(mat_size);
+	Vector previous_displacements(mat_size);
+	Vector displacements_copy(mat_size);
+	Vector displ_increment(mat_size);
+
+	rTangentTensor.resize(mat_size, mat_size);
+
+	// we fill the Vectors
+	for (unsigned int i = 0; i < number_of_nodes; i++) {
+		array_1d<double,3> displacement = rGeometry[i].FastGetSolutionStepValue(DISPLACEMENT);
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_elements/femdem2d_element.cpp]: You are not using references..., you can do it to save memory allocations

File: applications/FemToDemApplication/custom_elements/femdem2d_element.cpp [hunk 3]
```diff
@@ -1469,4 +1429,162 @@ void FemDem2DElement::AssignComponentsToSecondOrderCentralDifferencesTangentTens
...
+
+				damages[edge] = damage;
+			} // Loop over edges
+
+			// Calculate the elemental Damage...
+			const double damage_element = this->CalculateElementalDamage(damages);
+			const Vector& r_predictive_stress_vector = values.GetStressVector();
+			const Vector& r_integrated_stress_vector = (1.0 - damage_element) * r_predictive_stress_vector;
+
+			// std::cout << "**************************" << std::endl;
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_elements/femdem2d_element.cpp]: Remove if it will be not used on the future for debugging

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Some requests

**Comment by SergioJimenezReyes** [comment]
It is fine to me once you apply the changes that @loumalouomega suggested

**Review by SergioJimenezReyes — APPROVED** [review]
It is fine to me once you apply the changes that @loumalouomega suggested

**Review by loumalouomega — APPROVED** [review]

**Comment by AlejandroCornejo** [comment]
thank you @loumalouomega and @SergioJimenezReyes

**Pull request merged**